intranet/static/themes/snow/snow.js

```diff
@@ -5,426 +5,427 @@
 
 var snowroot = "/static/themes/snow/";
 
-if(navigator.appName == 'Microsoft Internet Explorer') {
-	var ie=true;
-} else {
-	var ie=false;
-}
-var mobile=false;
-if(navigator.userAgent.toLowerCase().indexOf("android") != -1) {
-	var android=true;
-	var mobile=true;
-} else {
-	var android=false;
-}
-var fastbrowser = false;
-if(!ie) fastbrowser = true;
+var ie = (navigator.appName === 'Microsoft Internet Explorer');
+
+var android = (navigator.userAgent.toLowerCase().indexOf("android") !== -1);
+var mobile = android;
+
+var fastbrowser = !ie;
+
 //Config
 //Number of flakes
 
-if(typeof snowmax != 'undefined') {
-
-} else if(mobile) {
-	var snowmax = 15;
-} else {
-	var snowmax = 100;
+if (typeof snowmax === 'undefined') {
+    var snowmax = mobile ? 15 : 100;
 }
 
-if(!window.requestAnimationFrame) {
-	window.requestAnimationFrame = function(f) {
-		return f();
-	}
+if (!window.requestAnimationFrame) {
+    window.requestAnimationFrame = function(f) {
+        return f();
+    }
 }
 
 //Colors possible for flakes
-var snowcolor=new Array("#aaaacc","#ddddFF","#ccccDD");
-// Number of snowflake characters in following array
-var numsnowletters=3;
+var snowcolor = ["#aaaacc", "#ddddFF", "#ccccDD"];
+//Number of snowflake characters in following array
+var numsnowletters = 3;
 //Fonts possible for flakes
-var snowtype=new Array("Arial Black","Arial Narrow","Times","Comic Sans MS");
-//Character to be used for flakes
-if(typeof snowletter != 'undefined') {
+var snowtype = ["Arial Black", "Arial Narrow", "Times", "Comic Sans MS"];
 
-} else if(!ie){ // IE doesnt' like it for some reason
-	var snowletter=new Array("❄","❅","❆");
-}else{
-	var snowletter="*";
+//Character to be used for flakes
+if (typeof snowletter === 'undefined') {
+    //IE doesnt' like it for some reason
+    var snowletter = ie ? "*" : ["❄", "❅", "❆"];
 }
-//Speed multiplier for the snow falling
-if(typeof sinkspeed != 'undefined') {
 
-} else if(fastbrowser) { // They have more elements and do piling. This increases the amount of time it takes for significant slowdown.
-	var sinkspeed=0.5;
-} else {
-	var sinkspeed=1;
+//Speed multiplier for the snow falling
+if (typeof sinkspeed === 'undefined') {
+    //They have more elements and do piling. This increases the amount of time it takes for significant slowdown.
+    var sinkspeed = fastbrowser ? 0.5 : 1;
 }
-if(typeof snowmaxsize != 'undefined' && typeof snowminsize != 'undefined') {
-
-} else if(mobile) {
-	//Maximum size of snowflakes
-	var snowmaxsize=44;
-	//Miniumum size of snowflakes
-	var snowminsize=16;
-} else {
-	//Maximum size of snowflakes
-	var snowmaxsize=22;
-	//Miniumum size of snowflakes
-	var snowminsize=8;
+
+if (typeof snowmaxsize === 'undefined' || typeof snowminsize === 'undefined') {
+    //Maximum size of snowflakes
+    var snowmaxsize = mobile ? 44 : 22;
+    //Miniumum size of snowflakes
+    var snowminsize = mobile ? 16 : 8;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['snowminsize' was used before it was defined.](https://www.codacy.com/app/tjcsl/ion/file/2746716172/issues/source?bid=3176160&fileBranchId=3355598#l51)